lxd/device/disk.go

```diff
@@ -431,34 +431,92 @@ func (d *disk) validateConfig(instConf instance.ConfigReader) error {
 			return fmt.Errorf("Storage volumes cannot be specified as absolute paths")
 		}
 
-		// Only perform expensive instance pool volume checks when not validating a profile and after
-		// device expansion has occurred (to avoid doing it twice during instance load).
-		if d.inst != nil && !d.inst.IsSnapshot() && len(instConf.ExpandedDevices()) > 0 {
+		var dbVolume *db.StorageVolume
+		var storageProjectName string
+
+		if d.inst != nil && !d.inst.IsSnapshot() && d.config["source"] != "" && d.config["path"] != "/" {
```

Comment:
This change is problematic because now it won't initialise `d.pool` for the root disk device.
This will then cause the `PreStartCheck` checks to be skipped for the root disk device (bad).

Reply:
wouldn't it be loaded anyway [here](https://github.com/hamistao/lxd/blob/issue12254/check_custom_block_storage_volume_sharing/lxd/device/disk.go#L496)?

Reply:
I'm struggling to follow the logic as there are lots of conditions at play. Lets break out the check and revisit.